errbot/backends/hipchat.py

```diff
@@ -346,6 +346,7 @@ class HipchatClient(XMPPConnection):
     def __init__(self, *args, **kwargs):
         self.token = kwargs.pop('token')
         self.endpoint = kwargs.pop('endpoint')
+        self._cached_users = None
```

Comment:
We have LRU cache we can use, there should be no need to implement it ourself, can you check why it doesn't work?
`@lru_cache(1024)`

Reply:
@gbin It does work, but not well enough to avoid a bunch of calls on start up. Reason for this is that each room participant is a unique user, even if it's same organizational user, so with 10 users per room, with 5 rooms, you would get 50 uncached calls. Each uncached call results in 2-3 API calls. users() gets all users, and since it's paged, this can (and does for us) results in multiple calls, depending on size of organization. It's not an enormous amount of calls, but large enough to trigger throttling on every start up.